### ISSUE TITLE
[Feature request] Stronger typing/dependency contract

### ISSUE THREAD

**Issue by apotonick** [trailblazer/trailblazer#169]
"contract.default"=> "contract"

class New < Trailblazer::Cell
  requires "model", "contract": Reform::Form # typing!
end

**Comment by konung** [comment]
Mental note: Possible solution. 

http://egonschiele.github.io/contracts.ruby/
https://github.com/egonSchiele/contracts.ruby

Ignore the unfortunate name “contracts” - this is not like Reform Contracts
this is type checking for method signatures - very similar to how Elm does it

**Comment by apotonick** [comment]
Correction: the name _contract_ for Reform was wrong, it should've been _form_. :stuck_out_tongue_winking_eye:

**Comment by mensfeld** [comment]
@apotonick I disagree. The name contract for reform is good as form implies that this is/can be a html form related stuff. The contract makes it more explicit and allows more junior people to get the mindshift of moving out of the AR validations scope.

**Comment by apotonick** [comment]
Ah sorry, I meant the name `Contract` _in Trailblazer_ was wrong, because we use it for HTML forms, e.g. `Form::Build` is really only needed to render a form (because of Reform's API).
